app/src/main/java/com/github/warnastrophy/core/ui/map/MapViewModel.kt

```diff
@@ -72,4 +82,77 @@ class MapViewModel(
       }
     }
   }
+
+  @SuppressLint("MissingPermission")
+  fun requestCurrentLocation(locationClient: FusedLocationProviderClient) {
+    viewModelScope.launch {
+      _uiState.value = _uiState.value.copy(isLoading = true)
+      val request =
+          CurrentLocationRequest.Builder()
+              .setPriority(Priority.PRIORITY_HIGH_ACCURACY)
+              .setMaxUpdateAgeMillis(0)
+              .build()
+
+      try {
+        // Use .await() to suspend the coroutine until the task is complete
+        val location = locationClient.getCurrentLocation(request, null).await()
+        println("Got location: $location !!")
+        location?.let {
+          val latLng = LatLng(it.latitude, it.longitude)
+          _uiState.value = _uiState.value.copy(target = latLng, isLoading = false)
+        }
+            ?: run {
+              // Handle the case where the successful task returns a null location
+              _uiState.value = _uiState.value.copy(isLoading = false)
+            }
+      } catch (e: Exception) {
+        // await() will throw an exception on failure
+        setErrorMsg("Error getting location: ${e.message}")
+        _uiState.value = _uiState.value.copy(isLoading = false)
+      }
+    }
```

Comment:
I suggest call `_uiState.value = _uiState.value.copy(isLoading = false)` once in a `finally` block so it's cleaner
```suggestion
    viewModelScope.launch {viewModelScope.launch {
    _uiState.update { it.copy(isLoading = true) }
    try {
        val request = CurrentLocationRequest.Builder()
            .setPriority(Priority.PRIORITY_HIGH_ACCURACY)
            .setMaxUpdateAgeMillis(0)
            .build()

        val location = locationClient.getCurrentLocation(request, null).await()
        location?.let {
            val latLng = LatLng(it.latitude, it.longitude)
            _uiState.update { it.copy(target = latLng) }
        } ?: setErrorMsg("Location unavailable")
    } catch (e: Exception) {
        setErrorMsg("Error getting location: ${e.message}")
    } finally {
        _uiState.update { it.copy(isLoading = false) }
    }
}
```